src/convert.cpp

```diff
@@ -665,16 +665,17 @@ namespace Exiv2 {
 
         if (subsecTag) {
             ExifData::iterator subsec_pos = exifData_->findKey(ExifKey(subsecTag));
-            if (   subsec_pos != exifData_->end()
-                && subsec_pos->typeId() == asciiString) {
-                std::string ss = subsec_pos->toString();
-                if (!ss.empty()) {
-                    bool ok = false;
-                    stringTo<long>(ss, ok);
-                    if (ok) subsec = std::string(".") + ss;
+            if (subsec_pos != exifData_->end()) {
```

Comment:
I'm not sure if I understand this change. Is it possible the query is doing something wrong here? 

To me, it seems that they are both doing the same thing, just that we now have two `if()` statements instead of one.

Reply:
I split `if (A && B) {...` into two nested ifs: `if (A) { if (B) {...`. It's because I want to move the call to `erase` inside the outer `if`, but not inside the inner `if`.

Reply:
Aha! sorry I missed that and thanks for the explanation :)